### ISSUE TITLE
Removing Buttons Block leaves a button inserter on the page.

### ISSUE THREAD

**Issue by paaljoachim** [WordPress/gutenberg#20821]
**Describe the bug**

Adding a buttons block. Then clicking the 3 dot drop down menu to remove the block only removes the button and leaves a trace (inserter) of the block on the page. Hovering over the inserter brings up the following.

<img width="750" alt="Screen Shot 2020-03-12 at 10 14 34" src="https://user-images.githubusercontent.com/5323259/76505786-4e2ab280-644a-11ea-811d-c12343ac73af.png">

I have to click to the right of the inserter to select the buttons block and then use the keyboard to delete it or use the 3 dot drop down menu and select Remove Block.
<img width="744" alt="Screen Shot 2020-03-12 at 10 15 39" src="https://user-images.githubusercontent.com/5323259/76505863-761a1600-644a-11ea-81e6-bfbf34b5e6b0.png">

**To reproduce**
1. Activate any of the default themes. Here is an example with Twenty Twenty.
2. Create a page.
3. Add a buttons block. Fill in with some button text.
4. Click the 3 dot drop down menu and select Remove Block.
5. Hover over the inserter and notice it says Add Button.

**Expected behavior**
Not having the inserter there at all.

**Screenshots**

Here is an example from Twenty Twenty. Page not saved.

![Delete-buttons-block](https://user-images.githubusercontent.com/5323259/76505625-16bc0600-644a-11ea-9757-00f972964dce.gif)

**Desktop (please complete the following information):**
Mac OSX 10.13.6
Chrome
Local site with Desktop Server.
Gutenberg version 7.7

<img width="1417" alt="Screen Shot 2020-03-12 at 10 07 50" src="https://user-images.githubusercontent.com/5323259/76505202-5a624000-6449-11ea-883a-52092d438d2e.png">

**Comment by youknowriad** [comment]
What you removed is the button block inside the buttons block. So in theory, the behavior is correct, that said, there might be ways to improve this. Remove the parent block when it contains no blocks or prevent removing all the inner blocks.

**Comment by paaljoachim** [comment]
Ahhh gotcha.
So we need somekind of condition.
Removing last button/single button also removes parent.
This also needs to be added to other similar child-parent blocks.

**For all nesting: Remove the last child block also removes parent block.**

**Comment by youknowriad** [comment]
> For all nesting: Remove the last child block also removes parent block.

That's not necessarily true, some blocks have other things shown like Cover / Media&Text and shouldn't be removed if the last inner block is removed. 

Even columns shouldn't be removed since it will show the placeholder. 

I think we should leave this issue as specific to the Button block.

**Comment by paaljoachim** [comment]
Yup good point!

**Comment by paaljoachim** [comment]
Closing this issue. I should rephrase it to suggest adding a placeholder for the Buttons Block.
